code/modules/holomap/holochip.dm

```diff
@@ -0,0 +1,269 @@
+GLOBAL_LIST_EMPTY(holochips)
+GLOBAL_LIST_EMPTY(holomap_cache)
+#define HOLOMAP_ERROR	0
+#define HOLOMAP_YOU		1
+#define HOLOMAP_OTHER	2
+#define HOLOMAP_DEAD	3
+
+/obj/item/clothing/accessory/holochip
+	name = "holochip"
+	desc = "A small chip, attachable to a jumpsuit, that allows for displaying a holographic map to the wearer."
+	var/destroyed = FALSE
+
+	//Holomap stuff
+	var/mob/living/carbon/human/activator
+	var/list/holomap_images
+	var/marker_prefix = "ert"
+	var/base_prefix = "ert"
+	var/holomap_color
+	var/holomap_filter = HOLOMAP_FILTER_ERT
+
+	var/list/prefix_update
+
+	var/obj/item/clothing/attached_to
+
+/obj/item/clothing/accessory/holochip/ui_action_click(mob/user)
+	if(!attached_to)
+		to_chat(world, "Attached to was [REF(attached_to)]")
+		return
+	togglemap(user)
+
+/obj/item/clothing/accessory/holochip/attach(obj/item/clothing/under/U, user)
+	. = ..()
+	if(.)
+		var/datum/action/A = new /datum/action/item_action/toggle_minimap(src)
+		if(ismob(U.loc))
+			A.Grant(user)
+		attached_to = U
+
+/obj/item/clothing/accessory/holochip/detach(obj/item/clothing/under/U, user)
+	deactivate_holomap()
+	for(var/datum/action/A in actions)
```

Comment:
What are you trying to do here. Just remove the action from the user instead of deleting it.